run/run.py

```diff
@@ -0,0 +1,348 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import configparser
+import gzip
+import json
+import platform as host_platform
+import re
+import requests
+import subprocess
+import sys
+import os
+
+"""
+run.py runs WPT and uploads results to Google Cloud Storage.
+
+The dependencies setup and running portion of this script should intentionally
+be left small. The brunt of the work should take place in WPT's `wptrun`:
+https://github.com/w3c/web-platform-tests/blob/master/tools/wptrun.py
+
+# Running the script
+
+Before you run the script, you need to:
+
+1. Copy run/running.example.ini to run/running.ini
+2. Modify the applicable fields of run/running.ini (this may also involve installing browsers)
+3. Make sure you have the correct secret in run/running.ini
+4. Install dependencies with `pip3 install -r requirements.txt`
+5. Make sure you have gsutil installed (see https://cloud.google.com/storage/docs/gsutil)
+
+The script will only accept platform IDs listed in browsers.json.
+
+By default this script will not upload anything! To do a full run for production:
+
+    ./run/run.py firefox-56.0-linux --upload --create-testrun
+
+# Filesystem and network output
+
+- This script will only write files under config['build_path']
+- One run will write approximately 111MB to the filesystem
+- If --upload is specified, it will upload that 111MB of results to Google Storage
+- To upload results, you must be logged in with `gcloud` and authorized
+"""
+
+def main(platform_id, platform, args, config):
+    assert sys.version_info.major == 3, 'This script requires Python 3.'
+
+    print('PLATFORM_ID:', platform_id)
+    print('PLATFORM INFO:', platform)
+
+    if args.path:
+        print('Running tests in path: %s' % args.path)
+    else:
+        print('Running all tests!')
+
+    if args.upload:
+        print('Setting up storage client')
+        from google.cloud import storage
+        storage_client = storage.Client(project='wptdashboard')
+        bucket = storage_client.get_bucket(config['gs_results_bucket'])
+        verify_gsutil_installed(config)
+
+    if args.create_testrun:
+        assert len(config['secret']) == 64, 'Valid secret required to create TestRun'
+
+    if platform['browser_name'] == 'chrome':
+        browser_binary = config['chrome_binary']
+        webdriver_binary = config['chromedriver_binary']
+    elif platform['browser_name'] == 'firefox':
+        browser_binary = config['firefox_binary']
+        webdriver_binary = config['geckodriver_binary']
+    else:
+        raise
```

Comment:
Might want to raise a more specific exception, or `assert False, "Unsupported browser"`.